src/interactable_044.c

```diff
@@ -0,0 +1,90 @@
+#include "global.h"
+#include "gba/types.h"
+
+#include "game.h"
+#include "interactable.h"
+#include "sprite.h"
+#include "task.h"
+
+#include "constants/interactables.h"
+#include "constants/move_states.h"
+
+typedef struct {
+    /* 0x00 */ SpriteBase base;
+} Sprite_IA044;
+
+#define sLeft   (screenX + ia->d.sData[0] * TILE_WIDTH)
+#define sRight  ((sLeft) + ia->d.uData[2] * TILE_WIDTH)
+#define sTop    (screenY + ia->d.sData[1] * TILE_WIDTH)
+#define sBottom ((sTop) + ia->d.uData[3] * TILE_WIDTH)
+void Task_Interactable_044(void)
+{
+    SpriteBase *object = TaskGetStructPtr(gCurTask);
```

Comment:
Is this required for match? So far I've always seen every Task have a unique struct ptr so probably shouldn't cast this to `SpriteBase`

Reply:
So should be

```
    Sprite_IA044 *ia044 = TaskGetStructPtr(gCurTask);
    SpriteBase *object = &ia044->base;
```